Library/Homebrew/cmd/migrate.rb

```diff
@@ -0,0 +1,22 @@
+require "migrator"
+require "formula_renames"
+
+module Homebrew
+  def migrate
+    raise FormulaUnspecifiedError if ARGV.named.empty?
+
+    # Don't do anything if name is a symlink
+    ARGV.named.each do |name|
+      name = $3 if name =~ HOMEBREW_TAP_FORMULA_REGEX
+      unless (rack = HOMEBREW_CELLAR/name).exist? && !rack.subdirs.empty?
+        raise NoSuchKegError, name
+      end
+      raise "#{rack} is a symlink" if rack.symlink?
+    end
```

Comment:
Ah, sure!

I thought about the case when we pass `newname` it can have `oldname` old name, but we can't know what was passed: `newname` or `oldname` since both of them are resolved as `newname`.